src/Cardano/Wallet.hs

```diff
@@ -49,10 +51,16 @@ import GHC.Generics
 import qualified Data.Set as Set
 
 -- | Types
-data WalletLayer m s = WalletLayer
-    { createWallet :: NewWallet -> ExceptT CreateWalletError m WalletId
-    , getWallet :: WalletId -> ExceptT GetWalletError m (Wallet s)
-    , watchWallet :: WalletId -> m ()
+data WalletLayer s = WalletLayer
+    { createWallet
+        :: NewWallet
+        -> ExceptT CreateWalletError IO WalletId
+    , readWallet
+        :: WalletId
+        -> ExceptT ReadWalletError IO (Wallet s, WalletMetadata)
+    , watchWallet
+        :: WalletId
+        -> IO ()
```

Comment:
So I've made this less polymorphic to make compilation simpler and error more friendly. In the end, we do not care about the monad parameter. And, if we ever need to test that on something else than IO (like `STM`), then, we'll do it at this moment.

Reply:
:+1: